publish.js

```diff
@@ -266,21 +266,37 @@ publish.tagVC = function (isTest, version, options) {
 };
 
 /**
- * Restore the package.json file to the latest committed version.
+ * Restore the files specified in filesToClean option to the latest committed version.
+ * Restores package.json and package-lock.json files to the latest commited version by default.
  *
- * Used internally to reset version number changes in package.json
- * @param moduleRoot {String} - the directory where the package.json file to
+ * Used internally to reset version number changes in list of files in filesToClean option.
+ * First checks whether the list of files in filesToClean option are being tracked by git or not.
+ * @param moduleRoot {String} - the directory where the files to
                                 clean is located in.
- * @param options {Object} - e.g. {"cleanCmd": "git checkout -- package.json"}
+ * @param options {Object} - e.g. {"checkFileTracking": "git ls-files --error-unmatch ${file}", "cleanCmd": "git checkout -- ${file}", "filesToClean": "package.json, package-lock.json"}
  */
 publish.clean = function (moduleRoot, options) {
     var originalDir = process.cwd();
 
     // change to the module root directory
     process.chdir(moduleRoot || "./");
 
-    // run the clean command
-    publish.execSyncFromTemplate(options.cleanCmd);
+    var file = options.filesToClean.split(/\s*,\s*/);
+
+    file.forEach(function (file) {
```

Comment:
For future reference, it is confusing and dangerous to reuse variable names in the way you've done here.  It would have been better to name the array `files`, for example.